notes/stv/20210822-validation-multi-user-01.txt

```diff
@@ -0,0 +1,335 @@
+#[root@ansibler]
+
+    ssh zeppelin
+
+        pushd /home/fedora/zeppelin
+
+            mv -b notebook \
+               notebook-origin
+
+	        git clone https://github.com/wfau/aglais-notebooks.git notebook
+
+	        bin/zeppelin-daemon.sh restart
+
+        popd
+    exit
+
+
+
+
+# -----------------------------------------------------
+# Install the 'jq' JSON parser.
+# https://github.com/wfau/aglais/issues/526
+#[root@ansibler]
+
+    ssh zeppelin \
+        '
+        sudo dnf -y install jq
+        '
+
+
+
+# -----------------------------------------------------
+# Login to our Zeppelin node and generate a new interpreter.json file.
+#[root@ansibler]
+
+    ssh zeppelin
+
+        # Create a list of notebooks
+        find /home/fedora/zeppelin/notebook -mindepth 1 -maxdepth 1 -type d ! -name '.git' -printf '%f\n' \
+        | tee /tmp/001.txt
+
+
+        # Create a JSON array of interpreter bindings.
+        sed '
+            1 i \
+"interpreterBindings": {
+        s/^\(.*\)$/"\1": ["spark", "md", "sh"]/
+        $ ! s/^\(.*\)$/\1,/
+        $ a \
+},
+            ' /tmp/001.txt \
+            | tee /tmp/002.txt
+
+
+        # Wrap our fragment as a JSON document to check
+        sed '
+            1 i \
+{
+        $ s/,//
+        $ a \
+}
+            ' /tmp/002.txt \
+        | jq '.'
+
+# -----------------------------------------------------
+# Truncate any existing list.
+#[user@zeppelin]
+
+        jq '
+            del(.interpreterBindings[])
+            ' \
+        /home/fedora/zeppelin/conf/interpreter.json \
+        > /tmp/003.json
+
+        sed -n '
+            /interpreterBindings/ p
+            ' /tmp/003.json
+
+
+
+
+# -----------------------------------------------------
+# Replace the empty list with our fragment.
+#[user@zeppelin]
+
+    # Insert our binding list into the rest of the file.
+    sed '
+        /interpreterBindings/ {
+            r /tmp/002.txt
+            d
+            }
+        ' /tmp/003.json \
+    | jq '.' \
+    > /tmp/004.json
+
+    # Run it through 'jq' to check.
+    jq '
+        .interpreterBindings
+        ' /tmp/004.json
+
+ 
+
+# -----------------------------------------------------
+# Replace the original interpreter.json from git.
+#[user@zeppelin]
+
+    mv /home/fedora/zeppelin/conf/interpreter.json \
+       /home/fedora/zeppelin/conf/interpreter.origin
+
+    cp /tmp/004.json \
+       /home/fedora/zeppelin/conf/interpreter.json
+
+    /home/fedora/zeppelin/bin/zeppelin-daemon.sh restart
+
+    >   Zeppelin stop                                              [  OK  ]
+    >   Zeppelin start                                             [  OK  ]
+
+
+
+# -----------------------------------------------------
+# Setup a tunnel to the monitor
+
+    ssh -L '3000:monitor:3000' fedora@128.232.227.178
+
+
+
+
+# -----------------------------------------------------
+# Setup a tunnel to the Yarn UI
+
+    ssh -L '8088:master01:8088' fedora@128.232.227.178
+
```

Comment:
Not clear which machine this is run on. I'm guessing this is a direct ssh connect from laptop to the zeppelin node.

Reply:
Yes it was run on local laptop, will be clearer in future notes